optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/cloner/gizmo/GizmoSolutionClonerFactory.java

```diff
@@ -0,0 +1,53 @@
+package org.optaplanner.core.impl.domain.solution.cloner.gizmo;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.optaplanner.core.api.domain.common.DomainAccessType;
+import org.optaplanner.core.api.domain.solution.cloner.SolutionCloner;
+import org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor;
+
+public class GizmoSolutionClonerFactory {
```

Comment:
Can be final.
I'd argue `final` is a useful keyword, signifying your intention for this class to not be overridden.